src/ragger/backend/speculos.py

```diff
@@ -105,12 +109,7 @@ def exchange_async_raw(self, data: bytes = b"") -> Generator[None, None, None]:
                                                p2=data[3],
                                                data=data[5:]) as response:
             yield
-            try:
-                self._last_async_response = RAPDU(0x9000, response.receive())
-            except ApduException as error:
-                if self.raises and not self.is_valid(error.sw):
-                    raise self._error(error.sw, error.data)
-                self._last_async_response = RAPDU(error.sw, error.data)
+            self._last_async_response = self.get_last_async_response(response)
```

Comment:
... following previous comment about `get_last_async_response`, if you don't want to create a specific fonction to trigger the decorator, you can use it directly as a fonction:
```python
self._last_async_response = RAPDU(0x9000, raise_policy_enforcer(response.receive)())
```

Reply:
I did not know that
I think however that the method with the @ decorator is more clear, but I'll change if you prefer calling the decorator as a function